### PR TITLE
chickenPackages: reduce brokenOnDarwin

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 225889` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>chickenPackages_5.chickenEggs.glls</li>
    <li>chickenPackages_5.chickenEggs.hypergiant</li>
    <li>chickenPackages_5.chickenEggs.noise</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>chickenPackages_5.chickenEggs.ezd</li>
    <li>chickenPackages_5.chickenEggs.gl-type</li>
    <li>chickenPackages_5.chickenEggs.gl-utils</li>
    <li>chickenPackages_5.chickenEggs.iconv</li>
    <li>chickenPackages_5.chickenEggs.scheme2c-compatibility</li>
  </ul>
</details>

File: pkgs/development/compilers/chicken/5/overrides.nix [hunk 1]
```diff
@@ -35,33 +35,46 @@ in {
...
   # git = addToBuildInputsWithPkgConfig pkgs.libgit2;
+  gl-utils = addPkgConfig;
   glfw3 = addToBuildInputsWithPkgConfig pkgs.glfw3;
+  glls = addPkgConfig;
+  iconv = addToBuildInputs (lib.optional stdenv.isDarwin pkgs.libiconv);
   icu = addToBuildInputsWithPkgConfig pkgs.icu;
   imlib2 = addToBuildInputsWithPkgConfig pkgs.imlib2;
-  lazy-ffi = old: (brokenOnDarwin old) // (addToBuildInputs pkgs.libffi old);
+  lazy-ffi = old:
+    # fatal error: 'ffi/ffi.h' file not found
```
Comment (Et7f3) [inline pkgs/development/compilers/chicken/5/overrides.nix:45]: libffi should provide it

Comment (nagy) [reply pkgs/development/compilers/chicken/5/overrides.nix:45]: libffi only provides `ffi.h` directly in the include folder without the `ffi/`  subfolder. The path would need to be patched.

Comment (wegank) [reply pkgs/development/compilers/chicken/5/overrides.nix:45]: Yeah, it should work with

```nix
  lazy-ffi = old:
    (addToBuildInputs pkgs.libffi old) // {
      postPatch = ''
        substituteInPlace lazy-ffi.scm \
          --replace "ffi/ffi.h" "ffi.h"
      '';
    };
```

**Pull request merged**